### PR TITLE
PR-5-Bookstore - Connect redux to api

### PR DESCRIPTION
**Pull request by ivanmvh**
## Learning objectives

- Understand the concept of state management.
- Use store, actions and reducers in React.
- Connect React and Redux.
- Connect an existing API via Redux and use the data to display something in a UI.

## Description

In this project, we will connect our React-Redux application to an existing API to create and remove books in a remote server.

### General requirements

- Make sure that there are no linter errors.
- Make sure that we used correct Gitflow.
- Make sure that we documented our work in a professional way.
- Follow our list of best practices for JavaScript.

### Project requirements

In the previous project we developed the *add book* and *remove book* features directly in our application using Redux and React state. Now we should refactor our code to use the *Bookstore API* to add and remove books.

- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- Use [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [`axios`](https://axios-http.com/docs/intro) with [`createAsyncThunk`](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
- Fetch book list from Bookstore API on Books page load.
    - Add an Action Creator, that returns a function.
    - Fetch list of books.
    - Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor our *add book* and *remove book* features to persist our changes in the server.
  - Change our existing Action Creators, so they return functions. 
  - Make changes by using API.
  - Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- *Add book* and *remove book* should work in the same way after the refactor.
- No styling is required.

@ivanmvh

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,8 @@
     "react-router-dom": "^6.4.3",
     "react-scripts": "5.0.1",
     "uuid": "^9.0.0",
+    "redux-thunk": "^2.4.2",
```
Comment (rustam-tolipov) [inline package.json:18]: - [ ] Great work on your project! I did notice, however, that there seems to be an issue with the link to the [MIT license](https://github.com/ivanmvh/bookstore-ivanmvh/blob/BR5-BOOKSTORE-CONNECT-REDUX-TO-API/MIT.md) in your README. Upon clicking on the link, it appears to redirect to a page that is not found. I kindly request that you take a moment to update the link, or if it has not yet been added, include it and personalize it to match the specifics of your project. Thank you for your attention to this matter.📝🔍🛠

Comment (rustam-tolipov) [inline package.json:18]: [OPTIONAL] You are free to remove this [line](https://github.com/ivanmvh/bookstore-ivanmvh/blob/BR5-BOOKSTORE-CONNECT-REDUX-TO-API/README.md#:~:text=NOTE%3A%20we%20recommend%20using%20the%20MIT%20license%20%2D%20you%20can%20set%20it%20up%20quickly%20by%20using%20templates%20available%20on%20GitHub.%20You%20can%20also%20use%20any%20other%20license%20if%20you%20wish.) as well.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @ivanmvh,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (ivanmvh) [reply package.json:18]: I also fix other things in the readme.md file.Thanks

Comment (ivanmvh) [reply package.json:18]: I don't undestand this remove line comment @Rustamxon7 . Thanks.

File: README.md [hunk 2]
```diff
@@ -255,28 +247,12 @@ I would like to thank Microverse community
...
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LICENSE -->
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./MIT.md) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+This project is [MIT license](https://choosealicense.com/licenses/mit/) licensed.
```
Comment (jerryowusu) [inline README.md]: Nice work on adding the license file, however, as stated by the previous reviewer, kindly ensure that you customize the license file by adding your name and year as shown below in the highlighted text. :point_down: 

Screenshot from 2023-01-27 06-36-47

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @ivanmvh , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close

 ## Highlights
Great effort so far :heavy_check_mark: 
All linters are passing :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (ivanmvh) [reply README.md]: So many time in Microverse and I didn't know these. Thanks!

**Review by topeogunleye — APPROVED** [review]
### Hi @ivanmvh, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**